src/java.base/unix/classes/sun/security/provider/NativePRNG.java

```diff
@@ -188,44 +188,46 @@ public RandomIO run() {
                         return null;
                     }
 
-                    try {
-                        return new RandomIO(seedFile, nextFile);
-                    } catch (Exception e) {
-                        return null;
-                    }
+                    return ThreadLocal.withInitial(() -> {
+                        try {
+                            return new RandomIO(seedFile, nextFile);
+                        } catch (Exception e) {
+                            return null;
+                        }
+                    });
                 }
         });
     }
 
     // return whether the NativePRNG is available
     static boolean isAvailable() {
-        return INSTANCE != null;
+        return INSTANCE.get() != null;
```

Comment:
I think this would be cleaner if it were `return INSTANCE() != null;`

i.e. replace the field with a getter. Existing variants return just the field, while thread-local calls `get()`.